src/Stream/ForwardSeekableReader.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "Reader.h"
+#include <cstdint>
+
+namespace Stream
+{
+	class ForwardSeekableReader : public Reader {
+	public:
+		// Get the size of the stream
+		virtual uint64_t Length() = 0;
+
+		// Get the current position of the stream
+		virtual uint64_t Position() = 0;
```

Comment:
I suppose technically the `Length` and `Position` attributes are independent of seeking. If we wanted we could have a `FiniteStream` class that has a well defined length and position, but no ability to seek. Not sure how granular we want to get though.